python-lib/spacy_tokenizer.py

```diff
@@ -167,25 +173,88 @@ def __init__(
                 Default is 10 million, higher than spaCy more conservative default at 1 million.
             add_pipe_components (list): List of spaCy pipeline components to add, for instance "sentencizer".
                 If use_models is False, only the tokenizer component is present so other components must be added explicitly.
-                If use_models is True, several pipeline components are automatically added. 
+                If use_models is True, several pipeline components are automatically added.
                 Please refer to the spaCy documentation to know which components are available for each model.
             enable_pipe_components (list, optional): List of spaCy pipeline components to enable.
-                To enable components, they must be added first, either by activating use_models 
+                To enable components, they must be added first, either by activating use_models
                 or by adding them explicitly in add_pipe_components.
             disable_pipe_components (list, optional): List of spaCy pipeline components to disable.
-                To disable components, they must be added first, either by activating use_models 
+                To disable components, they must be added first, either by activating use_models
                 or by adding them explicitly in add_pipe_components.
                 Please use either enable_pipe_components or disable_pipe_components, as both cannot be used at the same time.
 
         """
         store_attr()
         self.spacy_nlp_dict = {}
         self.tokenized_column = None  # may be changed by tokenize_df
+        self._restore_pipe_components = {}
+        """May be initialize in create_spacy_tokenizer as a spacy.language.DisabledPipes object. 
+        It is the disabled pipes from spacy.Languages.select_pipes() call, that can be restored by calling the object’s .restore() method."""
+
         if self.enable_pipe_components and self.disable_pipe_components:
             raise ValueError(
                 f"enable_pipe_components and disable_pipe_components are both non-empty. Please give either components to enable, or components to disable."
             )
 
+    def _set_use_models(self, languages: List[AnyStr]) -> bool:
+        """Set self.use_models attribute to True in case the text should be lemmatize with a SpaCy pre-trained model.
+        (e.g no lookups available for all languages in spacy-lookups-data)
+        This method should be called before creating your SpaCy Language object through create_spacy_tokenizer() method"""
+        if "pl" in languages or "ru" in languages:
+            self.use_models = True
+        else:
+            self.use_models = False
```

Comment:
I understand your comment, because those languages seems to come a little bit out of nowhere in the class. 
The 'languages' in _set_use_models will be the ones stored in spacy_nlp_dict, which is not filled in the constructor. The way it works now for my use is :
-you create a MultilingualTokenizer
-you call set_use_models with your list of languages 
- you call add_spacy_tokenizer on each language of your list

But I rather not add any language parameter in the init because it will break the way this class works, and as it is not mine (Alex C. wrote it for the nlp preparation plugin) I prefer not change it. Because the first purpose of this class was to call tokenize_df() with argument a 'language' or a 'language_column'  and not declare the languages in this init

Reply:
OK